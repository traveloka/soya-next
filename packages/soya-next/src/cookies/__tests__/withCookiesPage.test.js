import React from "react";
import { Cookies } from "react-cookie";
import withCookies from "../withCookiesPage";
import { render } from "@testing-library/react";

describe("withCookiesPage", () => {
  let Page, PageWithCookies;

  beforeEach(() => {
    Page = props => {
      expect(props).toMatchSnapshot();
      return (
        <>{props.init ? <div data-testid={"props-init"}>init</div> : <></>}</>
      );
    };
    Page.getInitialProps = () => ({ init: true });
    PageWithCookies = withCookies(Page);
  });

  afterEach(() => {
    delete process.browser;
  });

  describe("browser", () => {
    beforeEach(() => {
      process.browser = true;
    });

    it("should add cookie to page props", async () => {
      const props = await PageWithCookies.getInitialProps({});
      const { findByTestId } = render(
        <PageWithCookies {...props} cookies={new Cookies()} />
      );

      const propsInit = await findByTestId("props-init");
      expect(propsInit).toBeTruthy();
    });
  });

  describe("server", () => {
    beforeEach(() => {
      process.browser = false;
    });

    it("should add cookie to page props", async () => {
      global.MOCK_IS_NODE = true;
      const props = await PageWithCookies.getInitialProps({
        req: {
          universalCookies: new Cookies("key=value")
        }
      });
      const { findByTestId } = render(<PageWithCookies {...props} />);
      const propsInit = await findByTestId("props-init");
      expect(propsInit).toBeTruthy();
      delete global.MOCK_IS_NODE;
    });
  });
});
