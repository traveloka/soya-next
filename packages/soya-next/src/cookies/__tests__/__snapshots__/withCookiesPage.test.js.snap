// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCookiesPage browser should add cookie to page props 1`] = `
Object {
  "cookies": Cookies {
    "HAS_DOCUMENT_COOKIE": true,
    "changeListeners": Array [],
    "cookies": Object {},
  },
  "init": true,
}
`;

exports[`withCookiesPage server should add cookie to page props 1`] = `
Object {
  "cookies": Cookies {
    "HAS_DOCUMENT_COOKIE": true,
    "changeListeners": Array [],
    "cookies": Object {
      "key": "value",
    },
  },
  "init": true,
}
`;
